File: fix-imports.py
#!/usr/bin/env python3
"""
Script de correction automatique des imports pour compatibilité Vercel
Corrige motion/react, lucide-react@version, sonner@version, etc.
"""

import os
import re
from pathlib import Path

# Patterns de remplacement
REPLACEMENTS = [
    (r"from ['\"]motion/react['\"]", "from 'framer-motion'"),
    (r"from ['\"]lucide-react@[^'\"]*['\"]", "from 'lucide-react'"),
    (r"from ['\"]sonner@[^'\"]*['\"]", "from 'sonner'"),
    (r"from ['\"]framer-motion@[^'\"]*['\"]", "from 'framer-motion'"),
    (r"from ['\"]react-hook-form@[^'\"]*['\"]", "from 'react-hook-form'"),
]

def fix_file(filepath):
    """Corrige les imports dans un fichier"""
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original = content
        modified = False
        
        for pattern, replacement in REPLACEMENTS:
            if re.search(pattern, content):
                content = re.sub(pattern, replacement, content)
                modified = True
        
        if modified:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"✅ {filepath}")
            return 1
        return 0
    except Exception as e:
        print(f"❌ Erreur {filepath}: {e}")
        return 0

def main():
    """Parcourt et corrige tous les fichiers .tsx et .ts"""
    print("\n🔧 Correction des imports en cours...\n")
    
    files_fixed = 0
    ignored_dirs = {'.git', 'node_modules', '.next', 'dist', 'build', '.vercel'}
    
    for root, dirs, files in os.walk('.'):
        # Filtrer les dossiers à ignorer
        dirs[:] = [d for d in dirs if d not in ignored_dirs]
        
        for file in files:
            if file.endswith(('.tsx', '.ts', '.jsx', '.js')):
                filepath = os.path.join(root, file)
                files_fixed += fix_file(filepath)
    
    print(f"\n✨ {files_fixed} fichiers corrigés!\n")
    print("📋 Prochaines étapes:")
    print("   git add .")
    print('   git commit -m "fix: Correction imports pour Vercel"')
    print("   git push origin main\n")

if __name__ == "__main__":
    main()

File: fix-imports.js
#!/usr/bin/env node

/**
 * Script pour corriger automatiquement les imports de lucide-react et sonner
 * dans tous les fichiers .tsx
 * 
 * Usage: node fix-imports.js
 */

const fs = require('fs');
const path = require('path');

// Fonction récursive pour trouver tous les fichiers .tsx
function findTsxFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      // Ignorer node_modules et .git
      if (file !== 'node_modules' && file !== '.git' && file !== 'dist') {
        findTsxFiles(filePath, fileList);
      }
    } else if (file.endsWith('.tsx')) {
      fileList.push(filePath);
    }
  });
  
  return fileList;
}

// Fonction pour corriger les imports dans un fichier
function fixImports(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  let modified = false;
  
  // Corriger lucide-react sans version
  if (content.includes("from 'lucide-react'") || content.includes('from "lucide-react"')) {
    content = content.replace(/from ['"]lucide-react['"]/g, "from 'lucide-react@0.550.0'");
    modified = true;
  }
  
  // Corriger sonner sans version
  if (content.includes("from 'sonner'") || content.includes('from "sonner"')) {
    content = content.replace(/from ['"]sonner['"]/g, "from 'sonner@2.0.3'");
    modified = true;
  }
  
  // Corriger les versions incorrectes de lucide-react
  if (content.includes('lucide-react@0.562.0')) {
    content = content.replace(/lucide-react@0\.562\.0/g, 'lucide-react@0.550.0');
    modified = true;
  }
  
  // Corriger les versions incorrectes de sonner
  if (content.includes('sonner@1.0.0')) {
    content = content.replace(/sonner@1\.0\.0/g, 'sonner@2.0.3');
    modified = true;
  }
  
  // Sauvegarder si modifié
  if (modified) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`✅ Fixed: ${filePath}`);
    return true;
  }
  
  return false;
}

// Main
function main() {
  console.log('🔍 Searching for .tsx files...\n');
  
  const rootDir = process.cwd();
  const files = findTsxFiles(rootDir);
  
  console.log(`📁 Found ${files.length} .tsx files\n`);
  console.log('🔧 Fixing imports...\n');
  
  let fixedCount = 0;
  
  files.forEach(file => {
    if (fixImports(file)) {
      fixedCount++;
    }
  });
  
  console.log(`\n✨ Done! Fixed ${fixedCount} files`);
}

main();
